### PR TITLE
Prepare QuTiP 4.7.1 release.

### PR DESCRIPTION
**Pull request by hodgestar**
**Description**
Prepare QuTiP 4.7.1 release.

**Related issues or PRs**
- See #2045 for changelog update.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -105,16 +105,16 @@ jobs:
         os: [ubuntu-latest, windows-latest, macos-latest]
     env:
       # Set up wheels matrix.  This is CPython 3.6--3.10 for all OS targets.
-      CIBW_BUILD: "cp3{6,7,8,9,10}-*"
+      CIBW_BUILD: "cp3{6,7,8,9,10,11}-*"
       # Numpy and SciPy do not supply wheels for i686 for Python 3.10, so we
       # skip those too:
       CIBW_SKIP: "*-musllinux* cp310-manylinux_i686 cp310-win32"
```
Comment (Ericgig) [inline .github/workflows/build.yml:111]: We are skipping some build for 3.10. Do we need to do the same for 3.11?

Comment (hodgestar) [reply .github/workflows/build.yml:111]: Thanks! I need to check what CIBW currently builds against and what wheels numpy and scipy have now.

**Comment by hodgestar** [comment]
Build https://github.com/hodgestar/qutip/actions/runs/3675950003 succeeded. Now to check the results.

**Comment by hodgestar** [comment]
Reminder to myself that I will need to merge the changes made here into master.

File: .github/workflows/build.yml (hunk 1 above)
Comment (hodgestar) [reply .github/workflows/build.yml]: Builds appear to work nicely.

**Comment by hodgestar** [comment]
I've added #2047 to merge the build fixes into master.

**Pull request merged**